Breadcrumb docs (#474)
telerik/blazor-docs: components/breadcrumb/data-binding.md

---
title: Data Binding
page_title: Breadcrumb - Data Binding

> description: Data Binding basics in the Breadcrumb for Blazor.

Reviewer: ```suggestion
description: Data Binding basics for the Breadcrumb for Blazor.
```

slug: breadcrumb-data-binding
tags: telerik,blazor,breadcrumb,data,binding
published: True
position: 5
---

# Breadcrumb Data Binding

> This article explains how to provide data to a Breadcrumb component, so it renders items for you, the properties related to data binding and their results.

Reviewer: The sentence contains a subordinate clause ("so it renders items for you"), which breaks its flow of listed topics. We should either use a bullet list (which will repeat the list below), split the sentence to two, or make it simpler by removing "so it renders items for you".

Author: Simplified the sentence.

@[template](/_contentTemplates/common/general-info.md#valuebind-vs-databind-link)

This article has the following sections:

* The available (bindable) [features of a breadcrumb item](#breadcrumb-item-features).

> * How to match fields in the model with the Breadcrumb item [data bindings](#data-bindings).

Reviewer: ```suggestion
* How to map fields in the model to the Breadcrumb items [data bindings](#data-bindings).
```

* [Example](#example---data-binding-to-non-default-field-names) of using non-default field names for data binding.

## Breadcrumb Item Features

The Breadcrumb items provide the following features that you control through the corresponding fields in their data binding:

* `Text` - the text that will be shown on the item.
* `Title` - the text that will be added to the `title` attribute of the html element.
* `ImageUrl` / `Icon` / `IconClass`  - the URL to a raster image, the [Telerik icon]({%slug general-information/font-icons%}), or a class for a custom font icon that will be rendered in the item. They have the listed order of precedence in case more than one is present in the data (that is, an `ImageUrl` will have the highest importance). Read more in the [Icons article]({%slug breadcrumb-icons%}).
* `Url` - the view the item will navigate to by generating a link.
* `Disabled` -  you can disable items by setting this field to `true`. Such items will keep rendering but will not be clickable.
* `Class` - the CSS class that will be rendered on the main wrapping container of the item. You can use it to apply the desired styles to the separate Breadcrumb items.

## Data Bindings

> The properties of a Breadcrumb item match directly to a field of the model the Breadcrumb is bound to. You provide that relationship by providing the name of the field from which the corresponding information is present. To do this, use the properties in the main `TelerikBreadcrumb` tag:

Reviewer: ```suggestion
The properties of a Breadcrumb item map directly to fields from the Breadcrumb model. You define that relationship by providing the names of the fields where the corresponding information is present. To do this, use the properties in the main `TelerikBreadcrumb` tag:
```
